kubernetes/kubernetes.go

```diff
@@ -208,18 +208,36 @@ func (in *K8SClient) IsGatewayAPI() bool {
 		return false
 	}
 	if in.isGatewayAPI == nil {
-		isGatewayAPI := false
-		_, err := in.k8s.Discovery().RESTClient().Get().AbsPath("/apis/gateway.networking.k8s.io").Do(in.ctx).Raw()
-		if err == nil {
-			isGatewayAPI = true
-		} else if !errors.IsNotFound(err) {
-			log.Warningf("Error checking Kubernetes Gateway API configuration: %v", err)
-		}
+		isGatewayAPI := checkGatewayAPIs(in)
 		in.isGatewayAPI = &isGatewayAPI
 	}
 	return *in.isGatewayAPI
 }
 
+func checkGatewayAPIs(in *K8SClient) bool {
+	found := 0
+	res, err := in.k8s.Discovery().ServerResourcesForGroupVersion(K8sNetworkingGroupVersionV1.String())
+	if err != nil {
+		log.Debugf("Error while checking K8s Gateway API CRDs: %s. K8s Gateway API will not be used.", err.Error())
```

Comment:
Should this error message indicate which version we are expecting to find?  Or is that going to be in the Error string?